src/commands/apps-sdkconfig.ts

```diff
@@ -0,0 +1,123 @@
+import * as ora from "ora";
+import * as fs from "fs-extra";
+
+import * as Command from "../command";
+import {
+  AppMetadata,
+  AppPlatform,
+  getAppConfig,
+  getAppPlatform,
+  listFirebaseApps,
+} from "../management/apps";
+import { getOrPromptProject } from "../management/projects";
+import { FirebaseError } from "../error";
+import * as requireAuth from "../requireAuth";
+import * as logger from "../logger";
+import { promptOnce } from "../prompt";
+
+async function selectAppInteractively(
+  projectId: string,
+  appPlatform: AppPlatform
+): Promise<AppMetadata> {
+  if (!projectId) {
+    throw new FirebaseError("Project ID must not be empty.");
+  }
+
+  const apps: AppMetadata[] = await listFirebaseApps(projectId, appPlatform);
+  if (apps.length === 0) {
+    throw new FirebaseError(
+      `There are no ${appPlatform === AppPlatform.ANY ? "" : appPlatform + " "}apps ` +
+        "associated with this Firebase project"
+    );
+  }
+
+  const choices = apps.map((app: any) => {
+    return {
+      name:
+        `${app.displayName || app.bundleId || app.packageName}` +
+        ` - ${app.appId} (${app.platform})`,
+      value: app,
+    };
+  });
+
+  return await promptOnce({
+    type: "list",
+    name: "id",
+    message:
+      `Select the ${appPlatform === AppPlatform.ANY ? "" : appPlatform + " "}` +
+      "app to get the configuration data:",
+    choices,
+  });
+}
+
+module.exports = new Command("apps:sdkconfig [platform] [appId]")
+  .description(
+    "print the Google Services config of a Firebase app. " +
+      "[platform] can be IOS, ANDROID or WEB (case insensitive)"
+  )
+  .option("-o, --out [file]", "(optional) write config output to a file")
+  .before(requireAuth)
+  .action(
+    async (platform: string = "", appId: string = "", options: any): Promise<any> => {
+      let appPlatform = getAppPlatform(platform);
+
+      if (appPlatform === AppPlatform.PLATFORM_UNSPECIFIED) {
+        throw new FirebaseError("Unexpected platform. Only support iOS, Android and Web apps.");
```

Comment:
Since this error seems to appear quite frequently, would it be possible to store the error message as a constant somewhere?

Reply:
I'm throwing the error in `getAppPlatform()` helper function in `apps.ts` so this check is not necessary in these 3 files